File: Src/java/model-jackson/src/main/java/org/hl7/elm_modelinfo/r1/serializing/jackson/ModelInfoReaderProvider.java
package org.hl7.elm_modelinfo.r1.serializing.jackson;

import org.hl7.elm_modelinfo.r1.serializing.ModelInfoReader;

public class ModelInfoReaderProvider implements org.hl7.elm_modelinfo.r1.serializing.ModelInfoReaderProvider {
    @Override
    public ModelInfoReader create(String contentType) {
        if (contentType == null) {
            contentType = "application/xml";
        }

        switch (contentType) {
            case "application/xml": return new XmlModelInfoReader();
            default: throw new RuntimeException(String.format("ModelInfo reader content type %s not supported", contentType));
        }
    }
}

File: Src/java/model-xmlutil/src/main/java/org/hl7/elm_modelinfo/r1/serializing/xmlutil/ModelInfoReaderProvider.kt
@file:Suppress("detekt:all")

package org.hl7.elm_modelinfo.r1.serializing.xmlutil

import org.hl7.elm_modelinfo.r1.serializing.ModelInfoReader
import org.hl7.elm_modelinfo.r1.serializing.ModelInfoReaderProvider

class ModelInfoReaderProvider : ModelInfoReaderProvider {
    override fun create(contentType: String): ModelInfoReader? {
        var contentType: String? = contentType
        if (contentType == null) {
            contentType = "application/xml"
        }
        return when (contentType) {
            "application/xml" -> XmlModelInfoReader()
            else ->
                throw RuntimeException(
                    String.format("ModelInfo reader content type %s not supported", contentType)
                )
        }
    }
}
